src/vos/vos_io.c

```diff
@@ -1908,12 +1944,18 @@ dkey_update(struct vos_io_context *ioc, uint32_t pm_ver, daos_key_t *dkey,
 		goto out;
 	}
 
-	for (i = 0; i < ioc->ic_iod_nr; i++) {
-		iod_set_cursor(ioc, i);
+	if (krec->kr_bmap & KREC_BF_FLAT) {
```

Comment:
should we check vp_feat here? in case there are some corruption?

Reply:
I guess we could but it would just be to assert.